app/controllers/api/v1/recordings_controller.rb

```diff
@@ -57,19 +57,28 @@ def resync
         render_json(status: :ok)
       end
 
-      def publish_recording
-        publish = params[:publish]
+      # POST /api/v1/recordings/update_visibility.json
+      def update_visibility
         record_id = params[:record_id]
-        case publish
-        when 'true'
-          visibility = 'Published'
-        when 'false'
-          visibility = 'Unpublished'
+        visibility = params[:visibility]
+
+        case visibility
+        when 'Published'
+          published = true
+          protected = false
+        when 'Unpublished'
+          published = false
+          protected = false
+        when 'Protected'
+          published = true
+          protected = true
         end
+
         Recording.find_by(record_id:).update(visibility:)
-        BigBlueButtonApi.new.publish_recordings(record_ids: record_id, publish:)
+        BigBlueButtonApi.new.publish_recordings(record_ids: record_id, publish: published)
+        BigBlueButtonApi.new.update_recordings(record_ids: record_id, meta_hash: { meta_protected: protected })
```

Comment:
Is there a method we can use to edit the params and the metaparams of a recording?

Reply:
All of the `bbb_ruby_api` gem API methods are empowered by a general purpose method the `send_api_request`.
I've used it in my internship to create new API methods for custom requests, we can use it to create our own custom API requests.
Don't know if @farhatahmad would like this because will be tightly coupled to the implementation of a dependency.

Reply:
But, that's if the BBB API itself allows us to publish and set the recording `meta_data` at once because I think they are two different APIs `publish` vs `update`.